dubbo-registry/dubbo-registry-zookeeper/src/main/java/org/apache/dubbo/registry/zookeeper/ZookeeperRegistry.java

```diff
@@ -126,6 +151,41 @@ public void doUnregister(URL url) {
         }
     }
 
+    private void checkZkProvider(){
+        Set<URL> registeredURLSet = new HashSet<URL>(getRegistered());
+        for (URL registeredURL: registeredURLSet){
+            if(registeredURL.getProtocol().equalsIgnoreCase(Constants.CONSUMER))
+                continue;
+
+            String host = registeredURL.getHost();
+            int port = registeredURL.getPort();
+            String interfaceName = registeredURL.getServiceInterface();
+            List<String> childPath = zkClient.getChildren(Constants.PATH_SEPARATOR + DEFAULT_ROOT + Constants.PATH_SEPARATOR
+                    + interfaceName + Constants.PATH_SEPARATOR + Constants.PROVIDERS_CATEGORY);
+
+            boolean serviceExist = false;
+            for(String child : childPath) {
+                URL providerURL = URL.valueOf(URL.decode(child));
+                if(providerURL != null && providerURL.getHost().equalsIgnoreCase(host) && providerURL.getPort() == port){

```

Comment:
I think providerURL is never going to be null, if given child is null then URL.valueOf is going to throws a IllegalArgumentException exception.

Reply:
Just as an thought, can be rewritten as 

boolean serviceExist= childPath.stream()
                .map(URL::valueOf)
                .anyMatch(url->url.getHost().equalsIgnoreCase(host) && url.getPort()==port);